samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -41,4 +42,13 @@ public LocalApplicationRunner(Config config) {
     // 5. create the configuration for StreamProcessor
     // 6. start the StreamProcessor w/ optimized instance of StreamApplication
   }
+
+  @Override
+  public void kill(StreamApplication streamApp) {
+  }
```

Comment:
This whole class is unimplemented thus far. I figured if I added the exception here, whomever eventually implements the class may assume the exceptions were the appropriate behavior. 

Ultimately, all 3 methods need implementations and I think @xinyuiscool is planning to do it.